src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts

```diff
@@ -105,4 +117,16 @@ export class PagesWalletTransactionEditorComponent implements OnInit {
       return { amountIsZero: true };
     };
   }
+
+  private getCategoriesByType(typeToFilter: WalletTransactionType): void {
+    this.transactionCategoriesService.getCategories().subscribe((categoriesReceived) => {
+      this.transactionsCategories = categoriesReceived.filter(cat => cat.type == typeToFilter);
+    });
+  }
+  
+  private subscribeToFormValueChange(): void {
```

Comment:
This is not subscribing to form value changes but to single field value changes. Consider renaming to for example `listenToTypeChange`